core/cas-server-core-web/src/main/java/org/apereo/cas/config/CasCoreWebConfiguration.java

```diff
@@ -30,15 +35,37 @@
     @Autowired
     private CasConfigurationProperties casProperties;
 
+    /**
+     * Get messages from one or more property files containing values
+     * that should be exposed to Thyme templates but don't need i18n.
+     * Implementors could add custom-common-messages.properties to their overlay and
+     * keys in that will take precedence over the internal common-messages.properties.
+     * Keys in regular messages bundles will override any of the common messages.
+     * @return PropertiesFactoryBean containing all common (non-i18n) messages
+     */
+    @Bean
+    public PropertiesFactoryBean casCommonMessages() {
+        final PropertiesFactoryBean properties = new PropertiesFactoryBean();
+        final Resource[] propertyResources = new Resource[] {
```

Comment:
Since you have done this so well and now that we are here, could we move these somewhere into the configuration properties AND provide a default for a file that is NOT part of classpath? maybe in /etc/cas/config? 

Of course, document all?